Userland/Libraries/LibWeb/Streams/ReadableStream.h

```diff
@@ -30,7 +31,7 @@ class ReadableStream final : public Bindings::PlatformObject {
         Errored,
     };
 
-    static WebIDL::ExceptionOr<JS::NonnullGCPtr<ReadableStream>> construct_impl(JS::Realm&, Optional<JS::Handle<JS::Object>> const& underlying_source);
+    static WebIDL::ExceptionOr<JS::NonnullGCPtr<ReadableStream>> construct_impl(JS::Realm&, Optional<JS::Handle<JS::Object>> const& underlying_source, QueuingStrategy const& = {});
```

Comment:
Usually we wrap optional parameters with `Optional<T>`.

Reply:
hmm yeah, the reason why I didn't do that here is because the spec text  says:

> If strategy["size"] exists, throw a RangeError exception.

And
>     // 1. If strategy["highWaterMark"] does not exist, return defaultHWM.

Which to me kind of sounds like they are individually optional parameters of the struct?

Most of the other places it seems in our implementation that I can see which use as optional in the constructor also have the spec text somewhere of something like:

>     // 1. If form is given, then:

But Readable/Writeable Stream don't seem to have that wording, so I am thinking that if I make this optional, the code will end up looking slightly less close to spec text.

@kennethmyhra Do you think I am I maybe misunderstanding something here with the individually optional do you think?

Reply:
> If strategy["size"] exists, throw a RangeError exception.

That part threw me off a bit too. Personally I would probably have wrapped it in an `Optional<T>` thinking the spec wasn't verbose enough, but that's just me and I won't claim any authority on that 😆

Reply:
Yeah, so strategy["size"] is optional

Reply:
Yeah, so it seems strategy["size"] is optional, and so is strategy["highWaterMark"]. From reading the spec (may be misunderstanding something), it seems possible that size could be missing but high water mark is given?

I think that probably since both of the structs members are optional, I suppose it it the same thing as the entire thing being optional?

I actually did try making the entire thing optional, but to me it seemed a little bit more confusing to me with double 'exists' checks

Reply:
The bit of logic that I am thinking about specifically is:

```
        // 1. If strategy["size"] exists, throw a RangeError exception.
        if (strategy.size)
            return WebIDL::SimpleException { WebIDL::SimpleExceptionType::RangeError, "Size strategy not allowed for byte stream"sv };

        // 2. Let highWaterMark be ? ExtractHighWaterMark(strategy, 0).
        auto high_water_mark = TRY(extract_high_water_mark(strategy, 0));
```

Which indicates strategy["size"] is optional itself, followed by:

```
>     // 1. If strategy["highWaterMark"] does not exist, return defaultHWM.
    if (!strategy.high_water_mark.has_value())
        return default_hwm;:bn

   // more logic doing something with strategy.high_water_mark.value()
```

Which means that strategy["size"] could be missing, but strategy["highWaterMark"] could be there?

Reply:
Looked through the `extract_*` methods and the logic you are referring to here. I agree, it doesn't make sense for it to an Optional<T>.